File: dynamic_programming/1641_count_sorted_vowel_strings/1641_count_sorted_vowel_strings.go
package main

import "fmt"

func countVowelStrings(n int) int {
	dp := make([][5]int, n)

	for i := 0; i < 5; i++ {
		dp[0][i] = 1
	}

	for i := 1; i < n; i++ {
		for j := 0; j < 5; j++ {
			for k := 0; k <= j; k++ {
				dp[i][j] += dp[i-1][k]
			}
		}
	}

	count := 0

	for i := 0; i < 5; i++ {
		count += dp[n-1][i]
	}

	return count
}

func main() {
	n := 1
	res := countVowelStrings(n)
	fmt.Println(res)
}

File: dynamic_programming/1641_count_sorted_vowel_strings/1641_count_sorted_vowel_strings.py
class Solution:
    def countVowelStrings(self, n: int) -> int:
        dp=[1,1,1,1,1]
        if n==1:
            return 5

        for i in range(2,n+1):
            a = dp[0]
            e = dp[1]+dp[0]
            i = dp[0]+dp[1]+dp[2]
            o = dp[0]+dp[1]+dp[2]+dp[3]
            u = sum(dp)

            dp[0]=a
            dp[1]=e
            dp[2]=i
            dp[3]=o
            dp[4]=u
        
        return sum(dp)

n = 1
res = Solution().countVowelStrings(n)
print(res)